### DISCUSSION TITLE
How does this contact with the Display? ( Programming question )

### DISCUSSION THREAD

**Discussion by ftk789** [mathoudebine/turing-smart-screen-python#151] [General]
Hello, I've been thinking of making this in Javascript, I was wondering how does this repo contact with the Display? , is it only a Python thing? Or are there any DLL's that can be used to send informationS?

**Comment by mathoudebine** [comment]
Hello @ftk789 
This Python project communicates with the screen using a serial protocol
You can find the protocol details in the Python files used for display communication:
https://github.com/mathoudebine/turing-smart-screen-python/tree/main/library/lcd

The protocol is different between Turing smart screen (lcd_comm_rev_a.py) and XuanFang (lcd_comm_rev_b.py)

There is no DLL for now, only Python code, but it can be integrated to your own Python project if you want to control the display for other purposes than system monitoring: see here https://github.com/mathoudebine/turing-smart-screen-python/wiki/Control-screen-from-your-own-code

There is also this project that control the screen using C# code: https://github.com/usausa/turing-smart-screen

**Reply by ftk789** [reply]
Ahh ok that makes sense, So no DLL's, I was hoping it having a DLL so I can make a JS version out of it, But I'll try with the Python files ,Thank you VERY Much for the detailed answer!, And thank you for an amazing project like this.
